include/wil/registry_basic.h

```diff
@@ -0,0 +1,1602 @@
+            };
+
+            // reg_view with the raw HKEY type is non-owning
+            using reg_view_nothrow = reg_view_details::reg_view_t<::wil::err_returncode_policy>;
+
+#if defined(WIL_ENABLE_EXCEPTIONS)
+            // reg_view with the raw HKEY type is non-owning
+            using reg_view = reg_view_details::reg_view_t<::wil::err_exception_policy>;
+#endif // #if defined(WIL_ENABLE_EXCEPTIONS)
+        }
+
+        //
+        // open_*_key functions (throwing)
+        //
+#if defined(WIL_ENABLE_EXCEPTIONS)
+        inline ::wil::unique_hkey open_unique_key(HKEY key, _In_opt_ PCWSTR path, ::wil::reg::key_access access = ::wil::reg::key_access::read)
+        {
+            const reg_view_details::reg_view regview{ key };
+            return regview.open_unique_key(path, access);
+        }
+
+#if defined(__WIL_WINREG_STL)
+        inline ::wil::shared_hkey open_shared_key(HKEY key, _In_opt_ PCWSTR path, ::wil::reg::key_access access = ::wil::reg::key_access::read)
+        {
+            const reg_view_details::reg_view regview{ key };
+            return regview.open_shared_key(path, access);
+        }
+#endif // #if defined(__WIL_WINREG_STL)
+
+        //
+        // create_*_key functions (throwing)
+        //
+        inline ::wil::unique_hkey create_unique_key(HKEY key, _In_ PCWSTR path, ::wil::reg::key_access access = ::wil::reg::key_access::read)
+        {
+            const reg_view_details::reg_view regview{ key };
+            return regview.create_unique_key(path, access);
+        }
+
+#if defined(__SDDL_H__)
+        inline ::wil::unique_hkey create_unique_key(HKEY key, _In_ PCWSTR path, _In_opt_ PCWSTR security_descriptor, ::wil::reg::key_access access = ::wil::reg::key_access::read)
+        {
+            const reg_view_details::reg_view regview{ key };
+            return regview.create_unique_key(path, security_descriptor, access);
+        }
+#endif // #if defined(__SDDL_H__)
+
+#if defined(__WIL_WINREG_STL)
+        inline ::wil::shared_hkey create_shared_key(HKEY key, _In_ PCWSTR path, ::wil::reg::key_access access = ::wil::reg::key_access::read)
+        {
+            const reg_view_details::reg_view regview{ key };
+            return regview.create_shared_key(path, access);
+        }
+
+#if defined(__SDDL_H__)
+        inline ::wil::shared_hkey create_shared_key(HKEY key, _In_ PCWSTR path, _In_opt_ PCWSTR security_descriptor, ::wil::reg::key_access access = ::wil::reg::key_access::read)
+        {
+            const reg_view_details::reg_view regview{ key };
+            return regview.create_shared_key(path, security_descriptor, access);
+        }
+#endif // #if defined(__SDDL_H__)
+#endif // #if defined(__WIL_WINREG_STL)
+#endif // #if defined(WIL_ENABLE_EXCEPTIONS)
+
+        //
+        // open_key_nothrow functions
+        //
+        inline HRESULT open_key_nothrow(HKEY key, _In_opt_ PCWSTR path, _Out_ HKEY* hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT

```

Comment:
Nothrow versions of this should still own, IMO.

Reply:
If I do that they no one who uses this can use their own C++ types. I tried hard to find a balancing act here. returning an HKEY* plays great using wil::unique_hkey and wil::shared_hkey, because those types overload operator&